db/crud.go

```diff
@@ -1165,6 +1165,26 @@ func (db *DatabaseCollectionWithUser) PutExistingRevWithBody(ctx context.Context
 
 }
 
+func (db *DatabaseCollectionWithUser) SyncFnDryrun(ctx context.Context, body Body, docID string) (*uint32, string, base.Set, channels.AccessMap, channels.AccessMap, error) {
+	doc := &Document{
+		ID:    docID,
+		_body: body,
+	}
+	// Make sure the user has permission to modify the document before confirming doc existence
+	mutableBody, metaMap, newRevID, err := db.prepareSyncFn(doc, doc)
```

Comment:
Since we're getting 'body' and docID from the incoming request, I expected us to use the existing version of the document (from the bucket) as oldDoc, when it exists.   Otherwise I don't think there's any reason for callers to provide docID?